### PR TITLE
[MRG] DOC: Update contributing guide and windows install

### PR DESCRIPTION
**Pull request by ntolley**
More verbose description of the pull request process for new contributors.

### CODE REVIEW

File: doc/contributing.rst
Comment (jasmainak) [inline doc/contributing.rst:17]: I think if you use some of the wording from this section to make the next section a little more verbose, one could get rid of the section altogether.

File: doc/contributing.rst
Comment (rythorpe) [inline doc/contributing.rst:57]:
```suggestion
Making your first pull request
```

**Comment by jasmainak** [comment]
Not all the blocks of commands have rendered correctly. See: https://output.circle-artifacts.com/output/job/ec1d9318-79d4-4e21-8f6e-2916a48e8a06/artifacts/0/html/contributing.html

image

You might be missing a colon or two

**Comment by ntolley** [comment]
@jasmainak @rythorpe I updated the windows install instructions. The rendering issues on the contributing guide should be fixed now too

Besides a linkcheck error I need to resolve, I'm pretty set with this PR. Let me know if there's any changes you'd like to see!

File: README.rst
Comment (jasmainak) [inline README.rst:100]: I'm tempted to just nuke this from the instructions ... until we figure out a simpler solution.

File: README.rst
Comment (jasmainak) [inline README.rst:80]: provide a link to WSL installer?

File: README.rst
Comment (ntolley) [reply README.rst:100]: Alternatively, it acts as a pretty good deterrent against trying the Windows install ;)

**Comment by rythorpe** [comment]
I'll try to review in detail this evening!

**Comment by jasmainak** [comment]
merge button is yours @rythorpe !

File: README.rst
Comment (jasmainak) [reply README.rst:100]: we don't want to lose 25% of our customers:

image

from the pre-workshop survey

File: README.rst
Comment (rythorpe) [inline README.rst:88]: I thought the pip installation for Neuron works, but it's just that the mod files need to be build by calling them manually?

File: doc/contributing.rst
Comment (rythorpe) [inline doc/contributing.rst:91]:
```suggestion
8. Now push the feature branch to your fork::
```

File: doc/contributing.rst
Comment (rythorpe) [inline doc/contributing.rst:96]: Which link is supposed to show up on the terminal when you push? I'm not sure I follow what the point is here...

File: doc/contributing.rst
Comment (rythorpe) [inline doc/contributing.rst:109]: I've always wondered this: is the `--depth 1` argument necessary?

File: README.rst
Comment (ntolley) [reply README.rst:88]: Oh wow I think you're right, that is actually new. Do you have a windows machine you can verify this with?

File: README.rst
Comment (jasmainak) [inline README.rst:80]: It seems that it's not obvious Anaconda needs to be installed also on WSL. Clarifying may help!

File: doc/contributing.rst
Comment (ntolley) [reply doc/contributing.rst:96]: I think it's just to draw attention to the fact that an automatic message gets generated about opening a PR? It's not exactly this link (the link check will fail on the PR otherwise) but they serve the same purpose

File: doc/contributing.rst
Comment (ntolley) [reply doc/contributing.rst:109]: This just prevents downloading the entire commit history, it's still accessible if they do a git fetch

File: README.rst
Comment (rythorpe) [reply README.rst:88]: I was thinking that it was possible all along, but that's probably my ignorance talking. I could test it out tonight, but it might be prudent to merge this PR before then.

**Comment by jasmainak** [comment]
So just as an update, I met with two users on zoom today to debug their installation issues ... the Mac users got it installed in 15 minutes, for the Windows users I couldn't figure out the issue after 1.5 hours of debugging. WSL didn't help. All sorts of issues, related to display and also having the right ipython version. I ended up recommending Linux

File: doc/contributing.rst
Comment (rythorpe) [reply doc/contributing.rst:109]: Okay, that's what I thought. My super nitpicky suggestion would be to remove this if it's not necessary because simplicity is better, especially when trying to onboard a new contributor who is still trying to wrap their head around the git workflow.

**Comment by jasmainak** [comment]
@ntolley let me know when this is ready and you've addressed @rythorpe 's comments. Let's try to merge before the workshop

**Comment by ntolley** [comment]
@jasmainak I think it's best to put this back to WIP. I'd say the contributing guide is fairly low priority and not necessary to merge before the workshop. Ryan and I will troubleshoot the windows install on his machine when I return so we can have a solid set of install instructions

File: README.rst
Comment (jasmainak) [reply README.rst:88]: I don't think Neuron can be installed on Windows with pip ...

File: README.rst
Comment (jasmainak) [inline README.rst:102]: I don't think we should be encouraging regular users to do editable installs. You'll start to see users who messed up their local directory with git and then you have to teach them git :) The right fix is to update the MANIFEST.in and package_data directory correctly. The questions to answer are:

1. If you install Neuron on Windows, can it be installed within an Anaconda environment? See [here](https://web.archive.org/web/20201126015939/https://www.neuron.yale.edu/ftp/neuron/nrn_mswin_install.pdf)
2. If the mod files are built on Windows, what's the best way to copy the dll file to the Anaconda directory.

Step 2. is what doesn't get run in an editable install. We should fix it instead of suggesting users to do an editable install.

**Comment by jasmainak** [comment]
See my comments with regards to editable install. We need to do a "proper fix". But I'm okay to merge just the contributing guide and keeping a WIP PR open for the rest.

**Comment by ntolley** [comment]
@jasmainak @rythorpe I think I've addressed all the comments! Let me know if everything looks good to you once the docs merged. I just decided to say we don't support windows install for now. This might change by the time of the release but given our current experiences I think it's accurate to say for now.

**Comment by jasmainak** [comment]
I let @rythorpe merge if happy

**Review by rythorpe — APPROVED** [review]

**Comment by rythorpe** [comment]
Thanks @ntolley!

**Pull request merged**
